src/main/java/org/dependencytrack/persistence/ComponentQueryManager.java

```diff
@@ -348,6 +348,11 @@ public Component createComponent(Component component, boolean commitIndex) {
         final Component result = persist(component);
         Event.dispatch(new IndexEvent(IndexEvent.Action.CREATE, result));
         commitSearchIndex(commitIndex, Component.class);
+        //Update PriorityList with new Enabled/Disabled Sources to avoid conflicts
+        ConfigPropertyQueryManager configPropertyQueryManager = new ConfigPropertyQueryManager();
+        if(configPropertyQueryManager.isDedupEnabled()){
+            configPropertyQueryManager.updatePropertiesFromEnabledSources();
+        }
```

Comment:
I am seeking a method to update and retrieve the enabled sources before creating a new component. It would be ideal to keep this check in the component creation process to ensure that the most up-to-date data is used and to maintain the correct priority logic. 

Should I keep it there or should I move it somewhere else?